src/models/actionPlans.model.ts

```diff
@@ -0,0 +1,144 @@
+import db from "../database/db";
+import { Knex } from "knex";
+
+export interface ActionPlan {
+  user_id: string | number;
+  personal_rank: number;
+  professional_rank: number;
+  health_wellness_rank: number;
+  work_life_balance_rank: number;
+  motivation_rank: number;
+  relationships_rank: number;
+  personal_issues_field: string;
+  professional_issues_field: string;
+  decisions_field: string;
+  leadership_process_field: string;
+  key_action_items: string;
+}
+
+export interface ActionPlanRecord extends ActionPlan {
+  id: string | number;
+  created_at: Date;
+  updated_at: Date;
+  deleted_at?: Date | null;
+}
+
+export const actionPlanDataAccess = {
+  table: "action_plans",
+
+  async fetchActionPlansByUserId(
+    user_id: string | number,
+    date?: string
+  ): Promise<ActionPlanRecord[]> {
+    const actionPlans = await db(this.table)
+      .select()
+      .where({ user_id })
+      .whereNull("deleted_at")
+      .modify((qb: Knex.QueryBuilder) => {
+        if (date) {
+          qb.whereRaw(`created_at >= date_trunc('month', timestamp '${date}')`);
+          qb.whereRaw(
+            `created_at < date_trunc('month', timestamp '${date}') + interval '1 month'`
+          );
+        }
+      })
+      .catch((err) => {
+        throw new Error(err);
+      });
+
+    return actionPlans.map((actionPlan: ActionPlan) => {
+      return {
+        ...actionPlan,
+        key_action_items: JSON.parse(actionPlan.key_action_items),
+      };
+    });
+  },
```

Comment:
should this be in a service instead of the model?

Reply:
I've been putting axios/db calls in the model layer lately bc that's where the data is "modeled" (if we're being literal about it) and leaving the business logic to the service layer, BUT I do see that the all the other service files have the axios calls there so I don't wanna be the oddball... movin em 🤝

Reply:
Actually, I'm gonna go back on this, I have some reusable db function that lives in two service functions and rather than refactor I think it'd g2g and we can move onto other work. Hopefully this flow doesn't throw anyone off too much